### PR TITLE
Implement PhfHash for integral arrays of any size.

### PR DESCRIPTION
**Pull request by turbocool3r**
This might be useful, but I'm not sure if it's ok to force Rust 1.51+.

### CODE REVIEW

File: phf_shared/src/lib.rs [hunk 1]
```diff
@@ -364,62 +364,44 @@ impl PhfHash for char {
 }
 
 // minimize duplicated code since formatting drags in quite a bit
-fn fmt_array(array: &[u8], f: &mut fmt::Formatter<'_>) -> fmt::Result {
+fn fmt_array<T: std::fmt::Debug>(array: &[T], f: &mut fmt::Formatter<'_>) -> fmt::Result {
```
Comment (JohnTitor) [inline phf_shared/src/lib.rs:367]:
```suggestion
fn fmt_array<T: core::fmt::Debug>(array: &[T], f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

File: phf_shared/src/lib.rs [hunk 2]
```diff
@@ -364,62 +364,44 @@ impl PhfHash for char {
...
             fn phf_hash<H: Hasher>(&self, state: &mut H) {
-                state.write(self);
+                for v in self {
+                    v.phf_hash(state);
+                }
             }
         }
 
-        impl FmtConst for [$t; $n] {
+        impl<const N: usize>  FmtConst for [$t; N] {
```
Comment (JohnTitor) [inline phf_shared/src/lib.rs:382]:
```suggestion
        impl<const N: usize> FmtConst for [$t; N] {
```

The formatting is a bit broken, could you run rustfmt?

**Review by JohnTitor — APPROVED** [review]
That version is released on 2021-03-25, I think it's enough to bump MSRV with a minor version bump.

File: phf_shared/src/lib.rs (hunk 2 above)
Comment (turbocool3r) [reply phf_shared/src/lib.rs]: Sure, just fixed.

**Comment by JohnTitor** [comment]
Sorry for the delay, and thanks for the PR!

**Comment by turbocool3r** [comment]
Thanks for the review!

**Pull request merged**